### PR TITLE
Add basic cli tests for date/time options and fix build warnings

### PR DESCRIPTION
**Pull request by davidag**
Inspired by tests in #282 (which is sadly stalled), here are some basic tests for testing date/time correctness for all Watson commands which do have time-related options.

Besides that, a mock deprecation warning has been fixed by using mocker. It was a trivial replace, that's why I've decided to add it to this PR, although in a separate commit.

The message in the builds was as follows:
```
DeprecationWarning: "mock" fixture has been deprecated, use "mocker" instead
    '"mock" fixture has been deprecated, use "mocker" instead', DeprecationWarning

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

A pytest warning related to the datafiles plugin has also been fixed. The message was as follows:

```
.tox/py36/lib/python3.6/site-packages/_pytest/mark/structures.py:324

/home/travis/build/TailorDev/Watson/.tox/py36/lib/python3.6/site-packages/_pytest
/mark/structures.py:324: PytestUnknownMarkWarning: Unknown pytest.mark.datafiles - 
    is this a typo?  You can register custom marks to avoid this warning -
    for details, see https://docs.pytest.org/en/latest/mark.html

    PytestUnknownMarkWarning,
```

~~This PR also includes changes from PR #315 as they are necessary to run green. When that PR is merged, I will rebase and leave only the other commits, if that's OK for you :relaxed:~~

Update 2019-09-30: This PR is now ready to be code-reviewed / merged.

### CODE REVIEW

**Comment by davidag** [comment]
@jmaupetit Thanks to you merging #315 I've been able to remove the unneeded commit in this PR, and it should be ready to be code-reviewed / merged.

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+markers =
+    datafiles: pytest-datafiles plugin marker. This avoids warning message.
```
Comment (k4nar) [inline pytest.ini]: Refs https://github.com/omarkohl/pytest-datafiles/issues/8

**Comment by davidag** [comment]
> Thanks for taking the time to do that!

No problem!

> The way you structured the tests is a bit unusual, at least to my eyes. Are you following a known pattern, or is it just something you came up with? :)

I'm not sure I understand what you mean :sweat_smile: Is the hierarchy what looks unusual to you? I remember I programmed this trying to refactor code after adding each test in order to avoid duplicate code.

In any case, there's no pattern here, just what I came up with :relaxed: 

> No judgment involved, I would be happy to refactor it in what I think would be a more conventional way if you'd like.

Please, go ahead! I'm sure the tests will be more consistent and thorough if you could give me a hand here :+1:

**Comment by davidag** [comment]
@k4nar If it's of any help, I wouldn't mind refactoring the tests into a more usual structure if you pointed me to some example/reference code.

Thank you for your time and excuse my insistence!

**Comment by davidag** [comment]
@k4nar: I've removed inheritance to make the tests more "pytesty" :crossed_fingers: :relaxed:

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    ('2019-04-10T14:12:43'),
+]
+
+
+class OutputParser:
+    FRAME_ID_PATTERN = re.compile(r'id: (?P<frame_id>[0-9a-f]+)')
+
+    @staticmethod
+    def get_frame_id(output):
+        return OutputParser.FRAME_ID_PATTERN.search(output).group('frame_id')
```
Comment (k4nar) [inline tests/test_cli.py]: You can use `self.` :)
